### PR TITLE
feat!: replace StopMagic with autocompleteHeaders

### PR DESCRIPTION
**Pull request by theseion**
StopMagic is still supported for backwards compatibility.

### CODE REVIEW

**Comment by fzipi** [comment]
Ugh, maybe `AutoCompleteHeaders` being true as the default (like now), and then setting it to false?

**Comment by theseion** [comment]
That was my first impulse. However, there are no other fields that use values that differ from the type default yet, so I would have to switch to using constructors instead of `Input{}`. Might be a good idea, but I wanted to keep the PR small.

**Review by fzipi — COMMENTED** [review]
Change looks good. But, we cannot merge this removing stop_magic now.

Once we have version 1.1 or 2.0 of the test spec, maybe.

**Comment by theseion** [comment]
Ok. I'll rework it to retain `stop_magic` as legacy option.

**Comment by fzipi** [comment]
@theseion How is that rework going on? Need help on this?

**Comment by theseion** [comment]
If you want to, sure. I'm not getting much done at the moment.

**Comment by theseion** [comment]
I've rewritten the PR to use `*bool` for all values in `Input` and `Overrides`. That makes it possible to detect missing values. Because of that I've also been able to change the name (and inversen the logic) to `AutocompleteHeaders`. `StopMagic` still works as before but `AutocompleteHeaders` has precedence.

File: test/types.go [hunk 1]
```diff
@@ -66,9 +72,115 @@ type FTWTest struct {
...
 	} `yaml:"meta"`
 	Tests []Test `yaml:"tests"`
 }
+
+// ApplyInputOverride will check if config had global overrides and write that into the test.
+func ApplyInputOverrides(overrides *Overrides, input *Input) {
+	applySimpleOverrides(overrides, input)
+	applyDestAddrOverride(overrides, input)
+	applyHeadersOverride(overrides, input)
+	postProcessAutocompleteHeaders(overrides.AutocompleteHeaders, overrides.StopMagic, input)
```
Comment (fzipi) [inline test/types.go:87]: Way easier to read, thanks!

**Review by fzipi — APPROVED** [review]

**Pull request merged**